### DISCUSSION TITLE
Deploying nextJS portfolio to vercel. SMTP issue

### DISCUSSION THREAD

**Discussion by SalilMonga** [vercel/community#5385] [Help] [answered]
### Summary

I am trying to deploy a nextJS [portfolio](https://github.com/SalilMonga/SalilMonga.github.io) to Vercel. I am using nodemailer to send emails from the form. Everything works fine locally but when I deployed to Vercel I initially got a 500(internal server error). I found some documentation mentioning I need to use await promises in my API for SMTP which resolved that issue. But now I keep getting "page-0224932ba39a5281.js:1 POST https://salil-monga-github-io.vercel.app/api/contact 405 (Method Not Allowed)". Can anyone please help me out?
To reproduce, go to this https://salil-monga-github-io.vercel.app/ and look into console logs. This issue does not exist locally. 

### Steps to Reproduce

Open https://salil-monga-github-io.vercel.app/ and fill out the form to send a message. Look in the console logs to find this error.

**Comment by leerob** [comment]
Could you include the relevant code please?

**Reply by SalilMonga** [reply]
Hi @leerob, the relevant code is in this [file](https://github.com/SalilMonga/SalilMonga.github.io/blob/main/react-portfolio/pages/api/contact.js). I am using nodemailer to send the messages and this is the error I get.Sorry for the late reply! Thank you 
<img width="377" alt="Screenshot 2024-01-15 at 3 54 06 PM" src="https://github.com/vercel/community/assets/56597025/c81352ef-3430-4200-9487-84dc2cc24306">

**Reply by SalilMonga** [reply, marked as answer]
Hi @leerob, I was able to find the issue. I only had a local env file and never added those variables into Vercel. After adding them I was able to get everything working.
